Fix comment indent for slash-star comments with 'if' or 'for'

FormerPhysicist opened the pull request:
This pull request should fix the indentation issue associated with /* */ style comments that contain keywords like "if" or "for".

See #124

Wilfred commented on groovy-mode.el:
```diff
@@ -749,19 +749,43 @@ Then this function returns (\"def\" \"if\" \"switch\")."
+(defun groovy--extract-line-without-comments ()
+  "Extracts the part of the current line that is not a comment."
+  (let (code-text
+		(start-pos (line-beginning-position))
```
Would you mind indenting with spaces instead? That's what the rest of the file uses.

Wilfred commented on groovy-mode.el:
```diff
@@ -749,19 +749,43 @@ Then this function returns (\"def\" \"if\" \"switch\")."
+	(unless (groovy--comment-p start-pos)
+	  (save-excursion
+		(parse-partial-sexp start-pos end-pos nil nil nil t)
+		(setf code-text (buffer-substring start-pos (point)))
```
You don't need `setf` in these places, so please use `setq` instead.

Wilfred commented on groovy-mode.el:
```diff
@@ -749,19 +749,43 @@ Then this function returns (\"def\" \"if\" \"switch\")."
-        (throw 'done (buffer-substring (point) (line-end-position)))))))
-
+	(let (code-text)
+      (while t
```
Since this is inside the `let` it should be indented further. I like to use [aggressive-indent-mode](https://github.com/Malabarba/aggressive-indent-mode) to do this :)

FormerPhysicist commented:
@Wilfred I've fixed the the issues mentioned in your comments. I've also added some test cases to your unit test file. I think I added the test cases correctly but I wasn't able to figure out how to run them and confirm that. I tried M-x ert but it keeps saying that 0 tests were run....(?)

Anyways, let me know if there's anything else I need to do.  Thanks!
